app/controllers/media_controller.rb

```diff
@@ -21,6 +21,12 @@ def download
         elsif stream_from_s3?
           body = Services.cloud_storage.load(asset)
           send_data(body.read, **AssetManager.content_disposition.options_for(asset))
+        elsif proxy_via_nginx?
+          url = Services.cloud_storage.presigned_url_for(asset)
+          headers['X-Cloud-Storage-URL'] = url
+          headers['X-Cloud-Storage-Host'] = URI.parse(url).host
```

Comment:
It would be useful to explain in the commit note why you need *both* the full URL (including the host) *and* the host on its own.